contracts/DADBridgeWrapper.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.8.9;
+
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
+import { Pausable } from "@openzeppelin/contracts/security/Pausable.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+error InvalidToken(IERC20 token);
+error InvalidRecipient(address recipient);
+error InvalidAmount(uint256 amount);
+error UnsupportedOperation();
+
+contract DADBridgeWrapper is AccessControl, Pausable {
+    using SafeERC20 for IERC20;
+
+    bytes32 public constant ROLE_ADMIN = keccak256("ROLE_ADMIN");
+
+    IERC20 private immutable _token;
+
+    uint256 private _totalSupply;
```

Comment:
I think that `total supply` is a bit confusing here since this contract isn't (any kind of) a token. What about `totalMinted` instead?

Reply:
Plus, it doesn't even store the actual total supply of the associated token, but rather how much of that has been transferred out of this contract.
But as I understood from @yudilevi, this is the interface required by whoever we're doing this for (LiquidApps?).

Reply:
it's required by the bridge - it generally expect to work with tokens, not wrappers, and the total supply represents the total that was minted on that blockchain.